app/server.R

```diff
@@ -600,30 +635,59 @@ server <- function(input, output, session) {
 
   observeEvent(input$refreshColors, {
     COLOR_SEED(floor(runif(1, 1, 1000)))
-    output$summaryPlot <- renderPlotly({
-      summaryPlot()
-    })
-  })
+    RENDER_TRUTH <<- FALSE
+  },ignoreInit = TRUE)
 
   # When the target variable changes, update available forecasters, locations, and CIs to choose from
   observeEvent(input$targetVariable, {
-    CURRENT_WEEK_END_DATE(CASES_DEATHS_CURRENT)
-    if (input$targetVariable == "Deaths") {
-      df <- df %>% filter(signal == DEATH_FILTER)
-    } else if (input$targetVariable == "Cases") {
-      df <- df %>% filter(signal == CASE_FILTER)
+    #removing previous asofData
+    PREV_AS_OF_DATA(NULL)
+    
+    ## Defining Filter
+    if(input$targetVariable=="Deaths"){
+      FILTER = DEATH_FILTER
+    } else if (input$targetVariable=="Cases"){
+      FILTER = CASE_FILTER
     } else {
-      df <- df %>% filter(signal == HOSPITALIZATIONS_FILTER)
-      CURRENT_WEEK_END_DATE(HOSP_CURRENT)
+      FILTER = HOSPITALIZATIONS_FILTER
     }
-
+    
+    ## Filtering DF
+    df <- df %>% filter(signal == FILTER)
+    
+    ## Update available opitions
     updateAheadChoices(session, df, input$targetVariable, input$forecasters, input$aheads, TRUE)
     updateForecasterChoices(session, df, input$forecasters, input$scoreType)
     updateLocationChoices(session, df, input$targetVariable, input$forecasters, input$location)
     updateCoverageChoices(session, df, input$targetVariable, input$forecasters, input$coverageInterval, output)
-    updateAsOfData()
+    
+    ## setting to not call updateAsOfData
+    updateAsOf <- FALSE
+    
+    ## Resolving dates
+    if(PREV_TARGET %in%c("Deaths","Cases") && input$targetVariable%in%c("Deaths","Cases")){
+      CURRENT_WEEK_END_DATE(CASES_DEATHS_CURRENT)
+      currentFetch = input$asOf
+      if(input$asOf != "" &&
+         CURRENT_WEEK_END_DATE()!=currentFetch){
+        updateAsOf <- TRUE
+      }
+    }
+    else if(PREV_TARGET %in% c("Deaths","Cases") && input$targetVariable=='Hospitalizations'){
+      CURRENT_WEEK_END_DATE(HOSP_CURRENT)
+    }
+    else if(PREV_TARGET=='Hospitalizations' && input$targetVariable%in%c("Deaths","Cases")){
+      CURRENT_WEEK_END_DATE(CASES_DEATHS_CURRENT)
+    }
+    
+    ## Storing current target
+    PREV_TARGET <<- input$targetVariable
+    
+    if(updateAsOf){
+      updateAsOfData(fetchDate = currentFetch)
```

Comment:
Looks like `currentFetch` is only defined if the first conditional is triggered. Will this line raise an error if we're switching to/from hospitalizations?

Reply:
It won't raise an error because if we are switching from/to hospitalizations we won't need to call this function